patches/tModLoader/Terraria/WorldGen.cs.patch

```diff
@@ -2711,6 +2915,41 @@
  			int num = genRand.Next(4);
  			int num2 = 0;
  			int num3 = 0;
+@@ -50715,7 +_,11 @@
+ 					if (CountNearBlocksTypes(num11, num12, 2, 1, 27) > 0)
+ 						continue;
+ 
++					if (!TileLoader.Convert(num11, num12, 1)) {
++						if (genRand.Next(2) == 0)
++							flag2 = true;
```

Comment:
Hmm, the return value from `TileLoader.Convert` doesn't actually tell you if anything was converted. Perhaps we could check if `tile.type != preHookType`